src/Users.php

```diff
@@ -42,7 +42,7 @@ public function __construct(Silex\Application $app)
         $this->authtokentable = $prefix . "authtoken";
         $this->users = array();
         $this->session = $app['session'];
-        $this->remoteIP = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : "";
+        $this->remoteIP = $app['request']->getClientIp();
```

Comment:
The one we had troubles with specifically was with 'cookies' in the request object, I believe. But, In general, we should be really wary to change the few leftover `$_SERVER`'s, because in most cases there's a reason they're still there.